stdlib/Random/src/misc.jl

```diff
@@ -220,6 +220,20 @@ function shuffle!(r::AbstractRNG, a::AbstractArray)
     return a
 end
 
+function shuffle!(r::AbstractRNG, a::AbstractArray{Bool})
+    old_count = count(a)
+    uncommon_value = 2old_count <= length(a)
+    fuel = uncommon_value ? old_count : length(a) - old_count
```

Comment:
```suggestion
    len = length(a)
    uncommon_value = 2old_count <= len
    fuel = uncommon_value ? old_count : len - old_count
```
Somehow I doubt this will ever have a measurable performance impact, but I don't see how it could hurt.